linetools/isgm/abssightline.py

```diff
@@ -126,27 +120,24 @@ def __init__(self, radec, sl_type=None, em_type=None, comment=None, name=None):
 
         # Name
         if name is None:
-            self.name = 'J{:s}{:s}'.format(
-                    self.coord.ra.to_string(unit=u.hour,sep='',pad=True),
-                    self.coord.dec.to_string(sep='',pad=True,alwayssign=True))
+            self.name = ltu.name_from_coord(self.coord)
         else:
-            self.name=name
+            self.name = name
 
         # Others
         self.em_type = em_type
         self.sl_type = sl_type
+        self._abssystems = None
 
     def add_component(self, abscomp, tol=0.2*u.arcsec,
                       chk_sep=True, debug=False, **kwargs):
-        """Add an AbsLine object to the component if it satisfies
-        all of the rules.
+        """Add a component to AbsSightline if it satisfies all of the rules.
 
-        For velocities, we demand that the new line has a velocity
-        range that is fully encompassed by the component.
+        Presently, the only constraint is on RA/DEC
```

Comment:
I thought about this and decided that the source definition may vary from sub-class, e.g.
z for IGM sightlines, v or d for ISM sightlines.  Therefore decided to let the sub-class 
deal with this.